File: Python/pset6/mario/more/mario.py
from cs50 import get_int

def main():
    # initialize height variable
    height = 0

    # ask user for height between 1 and 8 using get_int until valid response is given
    while not 1 <= height <= 8:
        height = get_int("Height: ")

    # call function make to make pyramid by passing height twice
    # second height will remain constant
    make(height, height)

def make(row, height):
    # using second height as constant for height to keep track of how many spaces
    # are needed for each row on left side of pyramind

    # if row number is not 1 then call make on next row
    if row != 1:
        make(row-1, height)

    # draw left side of pyramid
    print(" " * (height-row), end="")
    print("#" * row, end="")

    # draw middle gap
    print(" " * 2, end="")

    # draw right side of pyramid
    print("#" * row)

main()

File: C/pset1/mario/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    //Get height that user wants for pyramid
    //height can be from 1 to 8 inclusively
    int ht;
    do
    {
        ht = get_int("Height: ");
    }
    while (ht < 1 || ht > 8);

    // i is being used to indicate which row we are creating. will create # of rows equal to height
    for (int i = 0; i < ht; i++)
    {
        //print the spaces needed to have correct indents on left side to make the ascending image of pyramid
        for (int j = ht - 1; j > i; j--)
        {
            printf(" ");
        }

        //print # needed to make left side of pyramid body
        //each row it will print an additonal # on left side
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }

        // print the double spaces for the middle of pyramid on each row
        printf("  ");

        //print # needed to make right side of pyramid body
        //each row it will print an additonal # on right side
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }

        //go down a line to start next line
        printf("\n");

    }
}